src/services/backgroundlayermgr.js

```diff
@@ -150,6 +162,36 @@ ngeo.BackgroundLayerMgr.prototype.set = function(map, layer) {
   return previous;
 };
 
+/**
+ * Return the current opacity background layer of a given map. `null` is returned if
+ * the map does not have an opacity background layer.
+ * @param {ol.Map} map Map.
+ * @return {ol.layer.Base} layer The opacity background layer.
+ * @export
+ */
+ngeo.BackgroundLayerMgr.prototype.getOpacityBgLayer = function(map) {
+  const mapUid = ol.getUid(map).toString();
+  return mapUid in this.mapUids_ ? this.ngeoLayerHelper_.getGroupFromMap(map, gmf.BACKGROUNDLAYERGROUP_NAME).getLayers().item(1) : null;
+};
+
+/**
+ * Set a background layer used by the opacity slider.
+ * @param {ol.Map} map The map.
+ * @param {ol.layer.Base} layer The opacity background layer.
+ */
+ngeo.BackgroundLayerMgr.prototype.setOpacityBgLayer = function(map, layer) {
+  layer.setOpacity(0);
+  layer.setVisible(true);
+  const bgGroup = this.ngeoLayerHelper_.getGroupFromMap(map, gmf.BACKGROUNDLAYERGROUP_NAME);
+  bgGroup.getLayers().insertAt(1, layer);
```

Comment:
You should remove all the extra layers (> 1) using a for (i=1; i<  bgGroup.getLayers().length) and bgGroup.getLayers().removeAt(i).

Then the method will be callable several times.